src/promptflow-devkit/promptflow/_sdk/_tracing.py

```diff
@@ -121,61 +124,96 @@ def _get_ws_triad_from_pf_config() -> typing.Optional[AzureMLWorkspaceTriad]:
 # and url with experiment should be printed at the beginning of experiment start.
 def _print_tracing_url_from_local(
     pfs_port: str,
-    collection: typing.Optional[str],
-    exp: typing.Optional[str] = None,
+    collection: str,
+    exp: typing.Optional[str] = None,  # pylint: disable=unused-argument
     run: typing.Optional[str] = None,
 ) -> None:
     url = f"http://localhost:{pfs_port}/v1.0/ui/traces/"
     if run is not None:
         url += f"?#run={run}"
-    elif exp is not None:
-        url += f"?#experiment={exp}"
-    elif collection is not None:
+    else:
+        # collection will not be None
         url += f"?#collection={collection}"
     print(f"You can view the traces from local: {url}")
 
 
 def _print_tracing_url_from_azure_portal(
-    ws_triad: typing.Optional[AzureMLWorkspaceTriad],
-    collection: typing.Optional[str],
-    exp: typing.Optional[str] = None,
+    ws_triad: AzureMLWorkspaceTriad,
+    collection: str,
+    exp: typing.Optional[str] = None,  # pylint: disable=unused-argument
     run: typing.Optional[str] = None,
 ) -> None:
-    if ws_triad is None:
+    # as this there is an if condition for azure extension, we can assume the extension is installed
+    from azure.ai.ml import MLClient
+
+    # we have different url for Azure ML workspace and AI project
+    # so we need to distinguish them
+    ml_client = MLClient(
```

Comment:
Considering local-to-cloud run also initilized a pfazure client (which contains a mlclient), feel like we can make trace and upload run share the same pfazure client? Not necessary to do it in this PR tho.

Reply:
i think it's kind of difficult to pass from tracing to run - this is my first feeling; it will be great to share this, let's discuss this later